frame/composable-traits/src/defi.rs

```diff
@@ -96,6 +96,22 @@ impl<AssetId: PartialEq> PartialEq for CurrencyPair<AssetId> {
 
 impl<AssetId: PartialEq> Eq for CurrencyPair<AssetId> {}
 
+impl<AssetId: Ord> Ord for CurrencyPair<AssetId> {
+	fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+		let base_ordering = self.base.cmp(&other.base);
+		if base_ordering.is_eq() {
```

Comment:
Seems this function would not be avoiding duplicates as it does not return "equal" for two pairs that are the swapped version of each other.  Because dex router supports reverse route iteration, we should consider swapped versions as equal no?

Reply:
So currently we don't store pairs as part of route, we just store pool_id and based on pair passed in exchange()/sell()/buy() operation we use all nodes in normal direction or reverse direction. So it is not possible to have route with say A-B and then same pool in reverse direction (B-A) at some point in route. So we don't need to worry about swapped version here.

To clarify more if we have a route BTC-ETH, ETH-USDT then I can exchange with two currency pair, either USDT-BTC (in that case it uses normal direction while performing any extrinsic on the route and if we do BTC-USDT then all operations on pool will be executed in reverse direction.